test/test_response.py

```diff
@@ -260,6 +260,16 @@ def test_decode_gzip_error(self) -> None:
         with pytest.raises(DecodeError):
             HTTPResponse(fp, headers={"content-encoding": "gzip"})
 
+    def test_decode_x_gzip(self) -> None:
```

Comment:
Since this test is identical to the above test could you instrad parametrize it?

Reply:
Sure thing, I parametrized `test_chunked_decoding_gzip()` and rebased.